xunit-autolabeler-v2/ast_parser/core/cli.py

```diff
@@ -36,12 +43,156 @@ def _write_output(output: List[str], output_file: str) -> None:
+        output_file: (Optional) A filepath to write the source file list to.
+                     Results will be written to stdout if this argument is
+                     omitted.
+    """
+    tested_files_filter = ShowTestedFilesOption.UNSPECIFIED
+    if show_tested_files == 'all':
+        tested_files_filter = ShowTestedFilesOption.ALL_TESTED
+    if show_tested_files == 'some':
+        tested_files_filter = ShowTestedFilesOption.ANY_TESTED
+    if show_tested_files == 'none':
+        tested_files_filter = ShowTestedFilesOption.NOT_TESTED
+
+    invocation = cli_list_source_files_datatypes.ListSourceFilesInvocation(
+        data_json,
+        root_dir,
+        tested_files_filter
+    )
+    result = cli_list_source_files.process_list_source_files(invocation)
+    output_lines = (
+        cli_list_source_files.format_list_source_files(invocation, result))
+
+    _write_output(output_lines, output_file)
+
+
+def inject_snippet_mapping(
+    data_json: str,
+    root_dir: str,
+    stdin_lines: List[str],
+    output_file: str = None
+) -> None:
+    """Adds snippet mapping to XUnit results
+
+    This method injects test-snippet mappings into XUnit test results provided
+    via stdin. It then saves the modified XUnit results to a file (if
+    output_file is specified) or prints them to stdout (if output_file is *not*
+    specified).
+
+    Args:
+        data_json: A path to a polyglot_drift_data.json file for the specified
+                   root directory
+        root_dir: A path to the target root directory.
+        stdin_lines: The lines of an XUnit test result file.
+        output_file: (Optional) A filepath to write the modified XUnit test
+                     output to. Modified XUnit output will be written to
+                     stdout if this argument is omitted.
+    """
+
+    grep_tags, source_tags, ignored_tags, source_methods = (
+        analyze.analyze_json(data_json, root_dir))
+
+    xunit_tree = etree.fromstring(''.join(stdin_lines))
+
+    for elem in xunit_tree.findall('.//testcase'):
+        class_parts = [part for part in elem.attrib['classname'].split('.')
+                       if not part.startswith('Test')]
+        test_key = (class_parts[-1], elem.attrib['name'])
+        for method in source_methods:
+            method_test_keys = [(
+                os.path.splitext(os.path.basename(test[0]))[0],
+                test[1]
+            ) for test in method['test_methods']]
+
+            if test_key in method_test_keys:
+                # Inject region tags into region_tags XML attribute
+                existing_tag_str = elem.attrib.get('region_tags')
+                existing_tag_list = (
+                    existing_tag_str.split(',') if existing_tag_str else [])
+
+                deduped_tag_list = (
+                    list(set(existing_tag_list + method['region_tags'])))
+
+                elem.set('region_tags', ','.join(deduped_tag_list))
```

Comment:
No need to make it a list again if you are just going to join it to a string, str.join take an iterable so set is OK:

deduped_tag_list = set(existing_tag_list + method['region_tags'])
elem.set('region_tags', ','.join(deduped_tag_list))

Reply:
Done.